### PR TITLE
Issue #2690325: Indentation incorrect on wrapped method calls

### PR DESCRIPTION
**Pull request by pfrenssen**
Ref https://www.drupal.org/node/2690325

### CODE REVIEW

File: coder_sniffer/Drupal/Sniffs/WhiteSpace/ObjectOperatorIndentSniff.php [hunk 1]
```diff
@@ -131,15 +135,23 @@ public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
...
 
                 // It can't be the last thing on the line either.
                 $content = $phpcsFile->findNext(T_WHITESPACE, ($next + 1), null, true);
                 if ($tokens[$content]['line'] !== $tokens[$next]['line']) {
                     $error = 'Object operator must be at the start of the line, not the end';
-                    $phpcsFile->addError($error, $next, 'LineStart');
+                    $fix   = $phpcsFile->addFixableError($error, $next, 'LineStart');
+                    if ($fix === true) {
+                        for ($i = ($next + 1); $i < $content ; $i++) {
+                            $phpcsFile->fixer->replaceToken($i, '');
```
Comment (klausi) [inline coder_sniffer/Drupal/Sniffs/WhiteSpace/ObjectOperatorIndentSniff.php]: I think this fix is not correct. The operator should be moved to the next line. We don't want to make lines even longer.

File: coder_sniffer/Drupal/Sniffs/WhiteSpace/ObjectOperatorIndentSniff.php [hunk 2]
```diff
@@ -131,15 +135,23 @@ public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
                                   $requiredIndent,
                                   $foundIndent,
                                  );
-                        $phpcsFile->addError($error, $next, 'Indent', $data);
+                        $fix   = $phpcsFile->addError($error, $next, 'Indent', $data);
```
Comment (klausi) [inline coder_sniffer/Drupal/Sniffs/WhiteSpace/ObjectOperatorIndentSniff.php]: This should be addFixableError(). So the fixer is never invoked here? Lol, why does the fixer even work in the test case??

File: coder_sniffer/Drupal/Test/WhiteSpace/ObjectOperatorIndentUnitTest.inc [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * @file
+ * Tests coding standards for chained method calls.
+ */
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+
+/**
+ * Function containing test code for chained method calls.
+ */
+function chained_method_calls(EntityTypeManagerInterface $entityTypeManager) {
```
Comment (klausi) [inline coder_sniffer/Drupal/Test/WhiteSpace/ObjectOperatorIndentUnitTest.inc]: Why is this function definition necessary? The sniff should also work in the global scope, right?

File: coder_sniffer/Drupal/Sniffs/WhiteSpace/ObjectOperatorIndentSniff.php (hunk 2 above)
Comment (pfrenssen) [inline coder_sniffer/Drupal/Sniffs/WhiteSpace/ObjectOperatorIndentSniff.php]: Indeed?!?

I had a brief look and `->addError()` also returns `TRUE` if the error is found, so this apparently works whether or not it has been marked as fixable. The only difference seems to be whether the log output reports it as fixable.

Well this is definitely fixable, will fix.

File: coder_sniffer/Drupal/Test/WhiteSpace/ObjectOperatorIndentUnitTest.inc (hunk 3 above)
Comment (pfrenssen) [inline coder_sniffer/Drupal/Test/WhiteSpace/ObjectOperatorIndentUnitTest.inc]: Putting it in a function ensures that the test covers this part of the code:

```
        $origConditions = null;
        if (isset($tokens[$stackPtr]['conditions']) === true) {
            $origConditions = $tokens[$stackPtr]['conditions'];
        }
        // ...
        while ($next !== false) {
            // ...
            $conditions = null;
            if (isset($tokens[$next]['conditions']) === true) {
                $conditions = $tokens[$next]['conditions'];
            }

            if ($origBrackets === $brackets && $origConditions === $conditions) {
                // ...
            }
        }
```

For completeness we should have at least one in the global scope too. Will move one test to the global scope and update documentation.

File: coder_sniffer/Drupal/Sniffs/WhiteSpace/ObjectOperatorIndentSniff.php (hunk 1 above)
Comment (pfrenssen) [inline coder_sniffer/Drupal/Sniffs/WhiteSpace/ObjectOperatorIndentSniff.php]: Good point, will fix!

**Pull request merged**